### PR TITLE
FOUR-15875 Optimize Script Tasks Execution with Nayra Engine

### PR DESCRIPTION
**Pull request by caleeli**
## Optimize Script Tasks Execution with Nayra Engine

image

image

## How to Test
- Go to the script executors
- Build and deploy the Nayra-PHP Executor
- Create a PHP-Nayra Script
- Add some php code to test
- Run it!

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-15875

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next
ci:deploy

### CODE REVIEW

File: upgrades/2024_06_12_150527_create_nayra_script_executor.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+use ProcessMaker\Models\ScriptExecutor;
+use ProcessMaker\ScriptRunners\Base;
+use ProcessMaker\Upgrades\UpgradeMigration as Upgrade;
+
+class CreateNayraScriptExecutor extends Upgrade
+{
+
```
Comment (julceslauhub) [inline upgrades/2024_06_12_150527_create_nayra_script_executor.php:9]: please delete this empty line

File: upgrades/2024_06_12_150527_create_nayra_script_executor.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        $exists = ScriptExecutor::where('language', Base::NAYRA_LANG)->exists();
+        if (!$exists) {
+            $scriptExecutor = new ScriptExecutor();
+            $scriptExecutor->language = Base::NAYRA_LANG;
+            $scriptExecutor->title = 'Nayra (µService)';
+            $scriptExecutor->save();
+        }
+    }
+}
```
Comment (julceslauhub) [inline upgrades/2024_06_12_150527_create_nayra_script_executor.php:25]: missing "down" method

File: upgrades/2024_06_12_150527_create_nayra_script_executor.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $exists = ScriptExecutor::where('language', Base::NAYRA_LANG)->exists();
+        if (!$exists) {
+            $scriptExecutor = new ScriptExecutor();
+            $scriptExecutor->language = Base::NAYRA_LANG;
+            $scriptExecutor->title = 'Nayra (µService)';
```
Comment (julceslauhub) [inline upgrades/2024_06_12_150527_create_nayra_script_executor.php:21]: Is this title is correct? I can see a strange character before "Service" word...

File: upgrades/2024_06_12_150527_create_nayra_script_executor.php (hunk 1 above)
Comment (caleeli) [reply upgrades/2024_06_12_150527_create_nayra_script_executor.php:9]: Fixed

File: upgrades/2024_06_12_150527_create_nayra_script_executor.php (hunk 2 above)
Comment (caleeli) [reply upgrades/2024_06_12_150527_create_nayra_script_executor.php:25]: Missing method was added

**Review by julceslauhub — APPROVED** [review]
approved

**Pull request merged**